examples/leasing/node_release_script.py

```diff
@@ -0,0 +1,187 @@
+#! /bin/python
+"""
+This script will free nodes from a project after
+a specific amount of time.
+
+You can find more information including
+config file and status file format in
+README file.
+"""
+
+import time
+import ConfigParser
+
+from hil.client.client import Client, RequestsHTTPClient
+from hil.client.base import FailedAPICallException
+
+
+class HILClientFailure(Exception):
+    """Exception indicating that the HIL client failed"""
+
+
+class StatusFileError(Exception):
+    """Exception regarding Status File Format"""
+
+
+def hil_client_connect(endpoint_ip, name, pw):
+    """Returns a HIL client object"""
+
+    hil_http_client = RequestsHTTPClient()
+    hil_http_client.auth = (name, pw)
+
+    return Client(endpoint_ip, hil_http_client)
+
+
+def load_node_info(statusfile):
+    """
+    Creates a structure containing all information from the
+    file. We will change and update this data structure and then
+    write it back to the file. In this way, the file will be read
+    and written once.
+    """
+    nodes = {}
+    with open(statusfile, 'r') as status_file:
+        for line in status_file:
+            node_status = line.split()
+            nodes[node_status[0]] = {}
+            nodes[node_status[0]].update({'project': node_status[1],
+                                          'time': node_status[2]})
+    return nodes
+
+
+def release_from_project(
+        hil_client, node, project
+        ):
+    """
+    We need to disconnect all the networks from all of the node's NICs
+    before releasing the node, after that we can detach node from
+    the project
+    """
+
+    node_info = hil_client.node.show(node)
+
+    for nic in node_info['nics']:
+        port = nic['port']
+        switch = nic['switch']
+        if port and switch:
+            try:
+                hil_client.port.port_revert(switch, port)
+                print('Removed all networks from node `%s`' % node)
+            except FailedAPICallException:
+                print('Failed to revert port `%s` on node \
+                        `%s` switch `%s`' % (port, node, switch))
+                raise HILClientFailure()
+
+    time.sleep(5)
+    # tries 2 times to detach the project because there might be a pending
+    # networking action setup (revert port in the previous step).
+    counter = 2
+    while counter:
```

Comment:
Would be a bit cleaner to use a for loop:

```python
for _ in range(2):
    ...
```

... and just use return inside the try, instead of break.